### PR TITLE
Update with-emotion to flush SSR styles on refresh

### PR DESCRIPTION
**Pull request by mattfysh**
This prevents styles which have changed between reloads from being duplicated in the output of `extractCritical`

cc @mitchellhamilton @hamlim @hawkins

### CODE REVIEW

**Comment by mitchellhamilton** [comment]
@mattfysh While you're at it, would you be able to update `emotion` and change [this line](https://github.com/zeit/next.js/blob/v3-beta/examples/with-emotion/pages/index.js#L51) to this?
```jsx
const Basic = styled.div`composes: ${basicStyles};`
```

**Comment by hawkins** [comment]
Good catch @mitchellhamilton , looks like I somehow missed that one in my pass through 😓

File: examples/with-emotion/pages/_document.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import Document, { Head, Main, NextScript } from 'next/document'
 import { extractCritical } from 'emotion/server'
+import { flush } from 'emotion'
+
+const dev = process.env.NODE_ENV === 'production'
```
Comment (mitchellhamilton) [inline examples/with-emotion/pages/_document.js]: Should this be `process.env.NODE_ENV !== 'production'`?

Comment (timneutkens) [inline examples/with-emotion/pages/_document.js]: 👍

**Comment by mattfysh** [comment]
Fixed :)

**Comment by timneutkens** [comment]
@mattfysh thanks! ❤️

**Review by timneutkens — APPROVED** [review]

**Pull request merged**